plugins/scaffolder-backend/src/scaffolder/stages/prepare/gitlab.ts

```diff
@@ -57,6 +57,7 @@ export class GitlabPreparer implements PreparerBase {
 
     const parsedGitLocation = parseGitUrl(location);
     const repositoryCheckoutUrl = parsedGitLocation.toString('https');
+    const ref = parsedGitLocation.toString('ref');
```

Comment:
I guess you can also use `parsedGitLocation.ref` directly.

Just a caveat that `parseGitUrl` is not great with branches containing `/`. :(

Reply:
@OrkoHunter Yeah, I thought that would work but then saw the example for `.https` and copied it. I'll simplify it.

Reply:
Thanks!